dspace-api/src/main/java/org/dspace/content/MetadataDSpaceCsvExportServiceImpl.java

```diff
@@ -113,25 +121,37 @@ public DSpaceCSV export(Context context, Community community, boolean exportAll)
     private Iterator<Item> buildFromCommunity(Context context, Community community)
         throws SQLException {
         Set<Item> result = new HashSet<>();
+        int itemsAdded = 0;
 
         // Add all the collections
         List<Collection> collections = community.getCollections();
         for (Collection collection : collections) {
-            Iterator<Item> items = itemService.findByCollection(context, collection);
-            while (items.hasNext()) {
+            // Never obtain more items than the configured limit
+            Iterator<Item> items = itemService.findByCollection(context, collection, getCsvExportLimit(), 0);
+            while (itemsAdded <= getCsvExportLimit() && items.hasNext()) {
                 result.add(items.next());
+                itemsAdded++;
```

Comment:
It would be better to use `result.size()` instead of a variable. Since it is a `Set`, if there are records mapped in multiple collections, they will only be included once.

Also, as a minor issue, it should be `<` instead of `<=` (an extra record is being included).